src/query/atomWithQuery.ts

```diff
@@ -162,8 +162,7 @@ export function atomWithQuery<
     (get, set, action: AtomWithQueryAction) => {
       switch (action.type) {
         case 'refetch': {
-          const { dataAtom, observer } = get(queryDataAtom)
-          set(dataAtom, new Promise<TData>(() => {})) // infinite pending
```

Comment:
@dai-shi I wish I knew why this fixes it, but it does. would be curious if you had any ideas

Reply:
I see this change now makes it impossible for atoms that reference a query atom to know that it's suspended, hence the failing test

Reply:
> I wish I knew why this fixes it

Me too.

Hmm, no, I don't have any ideas instantly. I'd need to dig into it.

Reply:
Your first approach would work with resolving the promise instead of infinite pending promise.
However, we should probably first need to reproduce the issue behavior in tests, otherwise we can't make sure to work in the future.